snap7/client_async.py

```diff
@@ -3,36 +3,131 @@
 """
 import asyncio
 import logging
-from ctypes import byref
+from ctypes import byref, Array
 
 from snap7.common import check_error
+from snap7.types import S7SZL
 from .client import Client
 
-logger = logging.getLogger(__name__)
+from concurrent.futures import ThreadPoolExecutor
+from functools import wraps, partial
+from typing import Union, Tuple, Awaitable
 
 
-def error_wrap(func):
-    """Parses a s7 error code returned the decorated function."""
-    async def f(*args, **kw):
-        code = await func(*args, **kw)
-        check_error(code, context="client")
+class ClientAsync(Client):
+    def __init__(self, max_workers=1):
+        super().__init__()
+        self.executor = ThreadPoolExecutor(max_workers=max_workers)
 
-    return f
+    def sync_to_async(self, func):
+        @wraps(func)
+        async def wrapper(*args, loop=None, executor=None, **kwargs):
+            if loop is None:
+                loop = asyncio.get_event_loop()
+            pfunc = partial(func, *args, **kwargs)
+            return await loop.run_in_executor(self.executor, pfunc)
+        return wrapper
 
+    async def as_compress(self, time) -> Awaitable[int]:
```

Comment:
I had a longer discussion once with @gijzelaerr about the topic where to store snap7-as_* methods.
The result was, that we left them in client.
I think it is possible to overload them in a Coroutine object without any trouble.

Reply:
I'll do as you say guys. Both options are good for me.